File: 01_Python/practice/문자열연습1/24_uniqueInOrder.py
# *******************************
# 연속 중복되지 않는 글자 출력
# 
    
data = [
    "all good tree",    # ['a', 'l', ' ', 'g', 'o', 'd', ' ', 't', 'r', 'e']
    "AAA AAAA AA AAA",  # ['A', ' ', 'A' ,' ', 'A', ' ', 'A']
    "i AM a BOY",       # ['i', ' ', 'A', 'M',' ', 'a', ' ', 'B', 'O', 'Y']    
]

def uniqueInOrder(sentence):

    result = []

    i = 0
    result.append(sentence[0])

    for j in range(1, len(sentence)):
        if sentence[i] != sentence[j]:
            result.append(sentence[j])
        
        i += 1

    return result
    
for s in data:
    print(uniqueInOrder(s))

# ↓ 결과 
# ['a', 'l', ' ', 'g', 'o', 'd', ' ', 't', 'r', 'e']
# ['A', ' ', 'A' ,' ', 'A', ' ', 'A']
# ['i', ' ', 'A', 'M',' ', 'a', ' ', 'B', 'O', 'Y']   

File: 03_Web/03_JavaScript/A_Basic/a35_문자열연습/24_uniqueInOrder.js
/*******************************
 * 
 */

const input = [
    "all good tree", // ['a', 'l', ' ', 'g', 'o', 'd', ' ', 't', 'r', 'e']
    "AAA AAAA AA AAA",  // ['A', ' ', 'A' ,' ', 'A', ' ', 'A']
    "i AM a BOY",  // ['i', ' ', 'A', 'M',' ', 'a', ' ', 'B', 'O', 'Y']

    // 다른 입력 데이터로 테스트 필요하면, 배열에 데이터 더 넣어보고 실행해도 됩니다
]

function uniqueInOrder(str) {
    arr = [];
    idx = 0;
    str.trim().split("").forEach(ch => {
        if (arr[idx - 1] != ch) { // 처음 undefined 와 비교 에러 x
            arr.push(ch) && idx++;
        }
    });

    return arr;
}

input.forEach(e => console.log(uniqueInOrder(e)));


// ueInOrder(str) {
//     arr = [];
//     idx = 0;
//     str.trim().split("").forEach(ch => {
//         if (idx) {
//             arr[idx - 1] != ch && arr.push(ch) && idx++;
//         } else {
//             arr.push(ch);
//             idx++;
//         }
//     });

//     return arr;
// }
// arr2 = [];

// console.log(arr2[-1])
// console.log(arr2[-1] == "a")